### PR TITLE
fix(`evm`): revert all revm changes

### PR DESCRIPTION
**Pull request by Evalir**
## Motivation

For some reason the revm upgrade broke a lot of stuff that shouldn't have been broken considering the migration was mostly transparent, and we've yet to triage the entirety of the issue.

## Solution

Revert back to `release/25` as a stable revm version. The current branch with latest revm changes is now `latest-revm`.

### CODE REVIEW

**Comment by mattsse** [comment]
sharing my preliminary findings,

for some reason the upgrade broke some requires

> HB:INVALID_WEIGHTS] invariant_fixedTermLoanManager_A() (runs: 1, calls: 23, reverts: 1)

which are caused by warps


```
  [0] DistributionHandler::distributorEntryPoint(1517936943072132424547585717294012335460981354448299090815164687 [1.517e63]) 
    ├─ [43820] 0x89CA9F4f77B267778EB2eA0Ba1bEAdEe8523af36::a6c4cff2(000000000003b09d4e7ad46bf4ed664cf7e3df863b40fc4a7f944ec95493210f) 
    │   ├─ [2519] DefaultsInvariants::currentTimestamp() [staticcall]
    │   │   └─ ← 1686326237 [1.686e9]
    │   ├─ [16831] VM::warp(1686326237 [1.686e9]) 
    │   │   └─ ← "HB:INVALID_WEIGHTS"
    │   ├─ [0] console::log(134087050498717332775286796463387467468078506391250064 [1.34e53]) [staticcall]
    │   │   └─ ← ()
    │   └─ ← 0x08c379a00000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000001248423a494e56414c49445f574549474854530000000000000000000000000000
    └─ ← ()
```

not sure why

File: crates/evm/src/executor/fork/cache.rs [hunk 1]
```diff
@@ -270,7 +265,7 @@ impl MemDb {
                 accounts.insert(add, acc.info);
 
                 let acc_storage = storage.entry(add).or_default();
-                if acc.status.contains(AccountStatus::Created) {
```
Comment (Evalir) [inline crates/evm/src/executor/fork/cache.rs:273]: While this should be OK, it seems that the `commit` function might need some tweaking, as it seems it wasn't recording the accounts as it should. We tried matching revm's commit fn but the issues still persisted.

File: crates/evm/src/fuzz/strategies/state.rs [hunk 2]
```diff
@@ -260,7 +260,7 @@ pub fn collect_created_contracts(
 
     for (address, account) in state_changeset {
         if !setup_contracts.contains_key(&b160_to_h160(*address)) {
-            if let (true, Some(code)) = (&account.is_touched(), &account.info.code) {
```
Comment (Evalir) [inline crates/evm/src/fuzz/strategies/state.rs:263]: This is the only part that really touches the fuzz strategies. It leads me to believe—maybe account recording is not being done correctly in some edge cases?

**Review by mattsse — APPROVED** [review]

File: crates/evm/src/fuzz/strategies/state.rs (hunk 2 above)
Comment (rakita) [reply crates/evm/src/fuzz/strategies/state.rs:263]: This "should" be fine, `is_touched` have the same functionality, and if it was wrong, bug would probably be a lot more serious and would pop on mainnet sync.

**Pull request merged**